### PR TITLE
[DOC-12610]: document `nprobe` tweaking

### PR DESCRIPTION
**Pull request by RayOffiah**
New page on fine-tuning the vector search.

### CODE REVIEW

File: modules/tutorials/partials/diagrams/couchbase-hierarchy.puml
Comment (sarahlwelton) [inline modules/tutorials/partials/diagrams/couchbase-hierarchy.puml:1]: Did this need to be included in this PR?

File: modules/vector-search/partials/nav.adoc [hunk 1]
```diff
@@ -1,6 +1,7 @@
 * xref:7.6@server:vector-search:vector-search.adoc[]
 ** xref:7.6@server:vector-search:create-vector-search-index-ui.adoc[]
 ** xref:7.6@server:vector-search:create-vector-search-index-rest-api.adoc[]
+** xref:7.6@server:vector-search:fine-tune-vector-search.adoc[]
```
Comment (sarahlwelton) [inline modules/vector-search/partials/nav.adoc]: I would move this below all of the Run a Vector Search pages in the TOC, personally.

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 2]
```diff
@@ -0,0 +1,132 @@
+= Fine-tuning Vector Searches
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
= Fine-Tuning a Vector Search Query
```

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+:page-topic-type: concept
+:page-ui-name: {ui-name}
+:page-product-name: {product-name}
+:stem:
+:description: pass:q[Additional parameters can be added to a vector search REST call that will allow you to tune the search for recall and/or accuracy.]
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
:description: Add additional parameters to a Vector Search REST API call to tune the search for recall or accuracy.
```

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 4]
```diff
@@ -0,0 +1,132 @@
+= Fine-tuning Vector Searches
+:page-topic-type: concept
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]: I'm not sure that this counts as a conceptual page.

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+:page-ui-name: {ui-name}
+:page-product-name: {product-name}
+:stem:
+:description: pass:q[Additional parameters can be added to a vector search REST call that will allow you to tune the search for recall and/or accuracy.]
+
+[abstract]
+{description}
+
+[#overview]
+== Overview
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]: If you're not linking to this from elsewhere, this isn't a useful heading. 

```suggestion

```

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+Couchbase automatically dynamically tunes vector indexes to achieve a balance between recall (the quality of the search results),
+latency (search response time), and memory efficiency.
+This tuning occurs during indexing and querying, eliminating the need for users to adjust optimization parameters manually.
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
The Search Service automatically tunes your Vector Search indexes to achieve a balance between: 

* Recall, or the quality of your search results
* Latency, or your search response time
* Memory efficiency

This tuning occurs during indexing and querying. 
You do not need to adjust these parameters manually. 
```

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+Couchbase dynamically adjusts two critical vector parameters:
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
Specifically, the Search Service dynamically adjusts two critical vector parameters:
```

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+
+`nlist` (or `Centroid` count)::
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
`nlist`, also known as `Centroid` count::
```

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+The number of clusters used for indexing.
+The centroids are used to quickly find the surrounding closest matches in the index. Increasing the number of centroids will increase accuracy but will decrease the speed of the search.
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
Centroids are used to quickly find the surrounding closest matches in the Vector Search index. 
Increasing the number of centroids will increase accuracy but will decrease the speed of the search.
```
Remember to use ventilated prose, please.

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
++
+The `nlist` is determined dynamically based on the size of the dataset (number of vectors) in a partition:
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
The `nlist` is determined dynamically based on the size of the dataset, or the number of vectors in a partition:
```

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
++
+[%header, cols="3*a"]
+|===
+| Number of vectors in partition (`nvec`)
+| `nlist` calculation
+| Notes
+
+| stem:["nvec " ge " 200,000"]
+| stem:[4 xx sqrt("nvec")]
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc:37]: Can you build a preview site, please? I want to see how this renders in the table.

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+ where increasing the number of datasets does not yield significant improvements in recall.
+
+| stem:["1000" lt "nvec" le "200,000" ]
+| stem:["nvec" / 100]
+| This formula targets approximately 100 vectors per cluster,
+which balances between too few and too many clusters, ensuring efficient indexing.
+
+| stem:["nvec" lt 1000]
+| N/A
+| For number of vectors less than 1000, Couchbase will carry out a straight forward one-to-one mapping between IDs and vectors with an exact vector comparison.
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
| For a number of vectors less than 1000, the Search Service will carry out a straight forward one-to-one mapping between IDs and vectors with an exact vector comparison.
```

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+Vectors are directly stored without the need for additional processing for the `nlist` calculation.
+
+|===
+
+`nprobes` (or `probes`)::
+The number of `centroids` that a query will check for similar vectors.
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
The number of `centroids` that a Search query will check for similar vectors.
```

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+The `nprobe` value is only set when Couchbase is using an Inverted File Index. Couchbase will select the best index type and comparison method depending on the size of the dataset and the `vector_index_optimized_for` setting.
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
The `nprobe` value is only set when the Search Service is using an Inverted File Index. The Search Service will select the best index type and comparison method depending on the size of the dataset and your `vector_index_optimized_for` setting.
```
Please add a link to where we explain the `vector_index_optimized_for` setting. I know it's discussed in creation topics.

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+| stem:[sqrt("nlist") / 2]
+| A minimum value of 1 is enforced to avoid setting `nprobe` too low.
+
+|===
+
+
+== Fine-Tuning Query Parameters
+
+
+The Vector Search API will accept an additional parameter object, allowing for tuning the recall and/or accuracy of the search execution.
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
Add an additional parameter object to any Vector Search query to tune the recall or accuracy of your search.
```

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 16]
```diff
@@ -0,0 +1,132 @@
...
+
+`ivf_nprobe_pct`::
+Adjusts the number percentage of clusters searched during queries, allowing for fine-tuning of the balance between recall and performance.
+If the value of `nprobe` is 5% of `nlist` (the centroid count), then setting the value of `ivf_nprobe_pct` higher than 5% will have the search cover a higher percentage of clusters, which will improve the accuracy of the search.
+
+`ivf_max_codes_pct`::
+The default value for this is 100, and the value represents the percentage of `centroids` that will be visited during a search. Reducing the value reduces the number of centroids visited, which will descrease accurary and recal, but will result in faster compute times.
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
You can add the following parameters to your query: 

`ivf_nprobe_pct`::
Adjusts the number percentage of clusters searched during queries, allowing for fine-tuning of the balance between recall and performance.
If the value of `nprobe` is 5% of `nlist` (the centroid count), then setting the value of `ivf_nprobe_pct` higher than 5% will have the search cover a higher percentage of clusters, which will improve the accuracy of the search.

`ivf_max_codes_pct`::
The default value for this is 100, and the value represents the percentage of `centroids` that will be visited during a search. 
Reducing the value reduces the number of centroids visited, which will decrease accuracy and recall, but will result in faster compute times.
```

These need to be added to the full Search documentation, with the explanation of all of the Search request parameters: https://docs.couchbase.com/cloud/search/search-request-params.html

If you don't feel up to doing that yourself, please log a ticket and I will pick it up.

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 17]
```diff
@@ -0,0 +1,132 @@
...
+
+For example, given a cluster `vector_index_optimized_for`: "recall" and `indexPartitions`: 5,  `nlist` and `nprobe` are determined based on the current vector count in a given partition:
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
== Example: nlist and nprobe Calculations on a Vector Search Index Set to Recall 

For example, if you had a Vector Search index with `vector_index_optimized_for` set to `"recall"` and `indexPartitions` set to `5`,  `nlist` and `nprobe` are determined based on the current vector count in a given partition:
```

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 18]
```diff
@@ -0,0 +1,132 @@
...
+    "k": 10,
+    "params": {
+      "ivf_nprobe_pct": 1,
+      "ivf_max_codes_pct": 0.2
+    },
+    "field": "embedding",
+    "vector": [0.024901132253900747, 1535]
+  }]
+}
+----
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]: How does this sample code relate to what's in the table above? The values provided are completely different from what you show in the calculation. What is this doing?

File: modules/vector-search/pages/run-vector-search-ui.adoc [hunk 19]
```diff
@@ -7,34 +7,34 @@
 [abstract]
 {description}
 
-For more information about how the Search Service scores documents in search results, see xref:run-searches.adoc#scoring[Scoring for Search Queries].
+For more information about how the Search Service scores documents in search results, see xref:fts:fts-scoring.adoc[Scoring for Search Queries].
```
Comment (sarahlwelton) [inline modules/vector-search/pages/run-vector-search-ui.adoc]:
```suggestion
For more information about how the Search Service scores documents in search results, see xref:search:run-searches.adoc#scoring[Scoring for Search Queries].
```

Wrong link. Please don't link to old search doc.

File: modules/vector-search/pages/run-vector-search-sdk.adoc [hunk 20]
```diff
@@ -9,57 +9,57 @@
 [abstract]
 {description}
 
-For more information about how the Search Service scores documents in search results, see xref:run-searches.adoc#scoring[Scoring for Search Queries].
+For more information about how the Search Service scores documents in search results, see xref:fts:fts-scoring.adoc[Scoring for Search Queries].
```
Comment (sarahlwelton) [inline modules/vector-search/pages/run-vector-search-sdk.adoc]:
```suggestion
For more information about how the Search Service scores documents in search results, see xref:search:run-searches.adoc#scoring[Scoring for Search Queries].
```
Wrong link. Please don't link to old search doc.

**Review by sarahlwelton — CHANGES_REQUESTED** [review]
Lot of noise in this PR with just deleting so many end-of-line spaces. I also have some concerns about how this is shaping up right now. Let me know if you want to discuss.

File: modules/vector-search/pages/fine-tune-vector-search.adoc (hunk 11 above)
Comment (RayOffiah) [reply modules/vector-search/pages/fine-tune-vector-search.adoc:37]: Yup, can do that.

https://preview.docs-test.couchbase.com/doc-12610/server/current/vector-search/fine-tune-vector-search.html

File: modules/tutorials/partials/diagrams/couchbase-hierarchy.puml
Comment (RayOffiah) [reply modules/tutorials/partials/diagrams/couchbase-hierarchy.puml:1]: I'll move it to a separate fix.

File: modules/search/pages/search-index-params.adoc [hunk 21]
```diff
@@ -1077,7 +1077,7 @@ The child field's type. Can be one of:
 
 For more information about the available field data types, see xref:field-data-types-reference.adoc[].
 
-|vector_index_optimized_for |String |Vector Only a|
+|[#vector-index-optimized-param]#vector_index_optimized_for# |String |Vector Only a|
```
Comment (sarahlwelton) [inline modules/search/pages/search-index-params.adoc:1080]: Why did we add the multiple hashes around this? I didn't think we needed to do that for table entries.

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 22]
```diff
@@ -0,0 +1,149 @@
...
+| For a number of vectors less than 1000, the Search Service will carry out a straight forward one-to-one mapping between IDs and vectors with an exact vector comparison.
+Vectors are directly stored without the need for additional processing for the `nlist` calculation.
+
+|===
+
+`nprobes` (or `probes`)::
+This is the number of `centroids` that a Search query will check for similar vectors.
+The `nprobe` value is only set when the Search Service is using an Inverted File Index. The Search Service will select the best index type and comparison method depending on the size of the dataset and your `vector_index_optimized_for` setting. 
++
+(For more information on the `vector_index_optimized_for` setting, see  xref:search:search-index-params.adoc#vector-index-optimized-param[Search Index JSON properties ])
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
For more information on the `vector_index_optimized_for` setting, see  xref:search:search-index-params.adoc#vector-index-optimized-param[Search Index JSON properties].
```

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 23]
```diff
@@ -0,0 +1,149 @@
...
+| Latency-optimized calculation (`vector_index_optimized_for: latency`)
+| stem:[sqrt("nlist") / 2]
+| A minimum value of 1 is enforced to avoid setting `nprobe` too low.
+
+|===
+
+==  Default `nlist` and `nprobe` calculations on a Vector Search Index
+
+
+The cluster maintains two dynamically adjusted parameters that will affect the speed, accuracy, and resources used during the search:
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
The cluster maintains two dynamically adjusted parameters that will affect the speed, accuracy, and resources used during a search: 
```

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 24]
```diff
@@ -0,0 +1,149 @@
...
+      "ivf_nprobe_pct": 1,
+      "ivf_max_codes_pct": 0.2
+    },
+    "field": "embedding",
+    "vector": [0.024901132253900747, 1535]
+  }]
+}
+----
+
+In the example above, the search will be carried out across all the fields, looking for close matches for the numbers specified in `vector`.
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
In the preceding example, the Search request returns results from any available fields in the index, but specifically searches the `embedding` field for the `10` closest matches to the vector `[0.024901132253900747, . . .]`.
The vector in the Search request has been truncated to reduce the display size of the example.
```

**Review by sarahlwelton — COMMENTED** [review]
Sorry, I'm being really picky but this is definitely shaping up really well :) Thanks for taking this on, Ray!

File: modules/vector-search/pages/fine-tune-vector-search.adoc [hunk 25]
```diff
@@ -0,0 +1,149 @@
...
+{
+  "fields": ["*"],
+  "knn": [{
+    "k": 10,
+    "params": {
+      "ivf_nprobe_pct": 1,
+      "ivf_max_codes_pct": 0.2
+    },
+    "field": "embedding",
+    "vector": [0.024901132253900747, 1535]
```
Comment (sarahlwelton) [inline modules/vector-search/pages/fine-tune-vector-search.adoc]:
```suggestion
    "vector": [0.024901132253900747, . . .]
```

**Review by sarahlwelton — CHANGES_REQUESTED** [review]
Sorry, actually saw the example you took this from. It's wrong in its current state. Please add this to the fixes, too.

**Review by sarahlwelton — APPROVED** [review]
Thanks @RayOffiah!

**Pull request merged**